source/js/ellipsis-util.js

```diff
@@ -0,0 +1,99 @@
+var ellipsis = require("html-ellipsis");
+var characterPerLine = require("./characters-per-line");
+
+var EllipsisUtil = {
+  /***
+   * @param {Object} lineStyles: Node with DOM elements that has a string
+   * @return {Boolean}
+   */
+  hasSpecificHeight: function(lineStyles) {
+    return lineStyles.height != "auto" && parseInt(lineStyles.height) != 0;
+  },
+
+  /***  
+   * @param {Node} element Node element.
+   * @return {Integer} Returns a font size.
+  * */
+  getFontSize: function(element) {
+    return parseInt(window.getComputedStyle(element).getPropertyValue("font-size"));
+  },
+
+  /***
+   * @summary Returns the width of the object 
+   * @param {Node} textLines: Node with DOM elements that has a string
+   * @return {Integer} width
+   */
+  getWidth: function(textLine) {
+    if(textLine.innerText === "") {
+      return 0;
+    }
+  
+    var lineStyles = window.getComputedStyle(textLine);
+
+    if(EllipsisUtil.hasSpecificHeight(lineStyles)) {
+      return parseInt(lineStyles.width);
+    }
+  },
```

Comment:
this function is not return anything it the "textLine.innerText " is not empty and has not ".hasSpecificHeight" .

Reply:
It would return undefined.

The cpl.calculate() is handling it. We **can** return 0, or undefined. 

It's the same for getParentsWidth.

Let's add errorhandling, because there is an actual error happening. Something we don't want. And it would still return undefined.

Reply:
0 would be the best as it says "@return {Integer}", abut the usage makes it even much stranger. There it seems to be used as a (callback?) function. "getStylesFunction", with is later uased as the content of "getWidth". Very confusing the naming

Reply:
If you want to test the output, set the hasSpecificHeight method and let it return false.

Reply:
So your answer is, that return undefined is right, or how should I understand you last comment?

My last point was, that I think we should return "0" as the description is saying it always returns an integer. Or we throw an error/exception if we can't determinate the height.

Reply:
Already done. As I said, you can test it with the above description. Thank you for paying attention to this.